### PR TITLE
Bug 1820401: daemon: better way to find installed kernel-rt packages on host

### PR DESCRIPTION
**Pull request by sinnykumari**
Initially we were relying on fixed set of kernel-rt
sub-packages which will be installed on the system. This can cause
issues when machine-os-content ships additional kernel-rt
sub-packages.

To solve this, we now query host with
`rpm-ostree status --json` and look for requested-local-packages
in the 0th index from available deployments list.

Fixes bug - https://bugzilla.redhat.com/show_bug.cgi?id=1820401

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -50,6 +50,39 @@ const (
...
+type Deployment struct {
+	RequestedLocalPkgs []string `json:"requested-local-packages"`
+}
+
+func installedRTKernelRpmsOnHost() ([]string, error) {
+	var out []byte
+	var err error
+	var rtKernelRpms = []string{}
+	if out, err = exec.Command("rpm-ostree", "status", "--json").Output(); err != nil {
+		wrappedErr := fmt.Errorf("Failed to execute rpm-ostre status --json %v", err)
```
Comment (runcom) [inline pkg/daemon/update.go:68]: You can directly return the fmt.Errorf below without the extra wrapperErr variable

**Comment by runcom** [comment]
Just a nit @cgwalters ptal as well if you have some time

/approve

File: pkg/daemon/update.go (hunk 1 above)
Comment (sinnykumari) [reply pkg/daemon/update.go]: Right, fixed it.

File: pkg/daemon/update.go [hunk 2]
```diff
@@ -715,11 +755,21 @@ func (dn *Daemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error
 
 	if canonicalizeKernelType(oldConfig.Spec.KernelType) == realtimeKernelType && canonicalizeKernelType(newConfig.Spec.KernelType) == realtimeKernelType {
 		if oldConfig.Spec.OSImageURL != newConfig.Spec.OSImageURL {
+			var installedRTKernelRpms []string
+			var err error
 			args = []string{"uninstall"}
-			args = append(args, rtKernel...)
+			if installedRTKernelRpms, err = installedRTKernelRpmsOnHost(); err != nil {
+				return fmt.Errorf("Error wihle fetching installed RT kernel on host %v", err)
```
Comment (ashcrow) [inline pkg/daemon/update.go:762]: `s|wihle|while|`

Comment (sinnykumari) [reply pkg/daemon/update.go]: damm type, fixed

File: pkg/daemon/update.go [hunk 3]
```diff
@@ -50,6 +50,38 @@ const (
...
+	}
+
+	var rpms HostInfo
+	if err := json.Unmarshal(out, &rpms); err != nil {
+		return rtKernelRpms, err
+	}
+
+	rtRegex := regexp.MustCompile("kernel-rt-.*")
+	for _, localPkg := range rpms.Deployments[0].RequestedLocalPkgs {
+		if rtRegex.MatchString(localPkg) {
```
Comment (cgwalters) [inline pkg/daemon/update.go:78]: This could just be `strings.HasPrefix(localPkg, "kernel-rt-")`.  But fine as is too.

File: pkg/daemon/update.go [hunk 4]
```diff
@@ -50,6 +50,38 @@ const (
 	realtimeKernelType = "realtime"
 )
 
+// HostInfo contains information of an OSTree based system
+type HostInfo struct {
```
Comment (cgwalters) [inline pkg/daemon/update.go:54]: We already have `rpmOstreeState` in `rpm-ostree.go`; would be good to share code there.

**Comment by sinnykumari** [comment]
Thanks Colin for review! Agree with you to unblock the CI test first. Will address the review comments in follow-up PR

**Comment by sinnykumari** [comment]
ugh, these failing ci tests

**Comment by runcom** [comment]
/lgtm

let's address Colin's address in a follow up!

**Comment by runcom** [comment]
maybe...

/lgtm

**Comment by sinnykumari** [comment]
Checked into live cluster from e2e-gcp-op CI test run, kernelType test went fine. e2-gcp-op test should finish fine

**Pull request merged**